cmdstanpy/model.py

```diff
@@ -921,18 +948,24 @@ def sample(
             else:
                 show_progress = show_progress and progbar.allow_show_progress()
 
-            get_logger().info('sampling: %s', runset.cmds[0])
+            runset = RunSet(
+                args=args,
+                chains=chains,
+                chain_ids=chain_ids,
+                time_fmt=time_fmt,
+                one_process_per_chain=one_process_per_chain,
+            )
             with ThreadPoolExecutor(max_workers=parallel_chains) as executor:
-                for i in range(chains):
+                for i in range(runset.num_procs):
```

Comment:
I don't love having this be unconditionally inside a threadpoolexecutor - isn't python going to spawn a bunch of threads (due to `parallel_chains`)?

IMO if we're only spawning one process we should avoid doing this

Reply:
does is spawn threads lazily?

Reply:
The docs seem to imply it is fine as is. It think the code is misleading to read however, as the spawning of threads and the loop are unnecessary in the (hopefully increasingly common) use case going forward

As an aside, should we define a 'minimum cmdstan version' we still support? E.g., at some point/version in the future we could just assume you have 2.28 or newer

Reply:
agreed, but how long should the window be?   I guess if we drop support for older versions of CmdStan in the future, then users can pin the CmdStanPy version to an earlier version of the package.

Reply:
I think something like “the previous 3 versions from the time of CmdStanPy release”

so 1.0 for example would support 2.25+, and if we released a 1.1 following 2.29, it would only guarantee support for 2.26+ 

3 is of course an arbitrary number. Right now RStan is so far behind that it’s worth supporting as old as we can reasonably manage if people will be working in multiple ecosystems

Reply:
inasmuch as we have the "force_one_process_per_chain" option, we need this logic and it works for all cases.